File: battle/data/MaskAtk.java
package common.battle.data;

import common.util.BattleStatic;
import common.util.Data;
import common.util.Data.Proc;
import common.util.unit.Trait;

import java.util.ArrayList;

public interface MaskAtk extends BattleStatic {

	default int getAltAbi() {
		return 0;
	}

	int getAtk();

	default int getDire() {
		return 1;
	}

	int getLongPoint();

	default int getMove() {
		return 0;
	}

	ArrayList<Trait> getATKTraits();

	Proc getProc();

	int getShortPoint();

	default int getTarget() {
		return Data.TCH_N;
	}

	boolean isOmni();

	boolean isRange();

	default int loopCount() {
		return -1;
	}

}

File: battle/data/MaskAtk.kt
package common.battle.data

import common.util.BattleStatic
import common.util.Data
import common.util.Data.Proc

interface MaskAtk : BattleStatic {
    fun getAltAbi(): Int {
        return 0
    }

    fun getAtk(): Int
    fun getDire(): Int {
        return 1
    }

    fun getLongPoint(): Int
    fun getMove(): Int {
        return 0
    }

    fun getProc(): Proc?
    fun getShortPoint(): Int
    fun getTarget(): Int {
        return Data.Companion.TCH_N
    }

    fun isRange(): Boolean
    fun loopCount(): Int {
        return -1
    }
}
